### PR TITLE
Update recommended mypy & CI to 1.1.1

### PR DESCRIPTION
**Pull request by intgr**
* Updated mypy to 1.1.1 for `compatible-mypy` and CI
* Minor tweak needed to typecheck ignores

## Related issues

* django-stubs PR: https://github.com/typeddjango/django-stubs/pull/1391
* Closes #373

### CODE REVIEW

File: requirements.txt
Comment (intgr) [inline requirements.txt:10]: TODO revert when django-stubs is released before merging this!

```suggestion
django-stubs==1.16.0
```

**Review by sobolevn — APPROVED** [review]
Yeap, or we can wait for a new `django-stubs` release before merging this.

**Comment by intgr** [comment]
Ooooops, before my morning coffee, I forgot I was supposed to wait for django-stubs release. Oh well, I can fix it later :sweat_smile:

**Pull request merged**
